core/deployment/src/main/java/io/quarkus/deployment/dev/filesystem/QuarkusFileManager.java

```diff
@@ -39,12 +41,22 @@ public void reset(Context context) {
                 this.fileManager.setLocation(StandardLocation.SOURCE_OUTPUT, List.of(context.getGeneratedSourcesDirectory()));
             }
             if (context.getAnnotationProcessorPaths() != null) {
+                // Paths might be missing! (see: https://github.com/quarkusio/quarkus/issues/42908)
+                ensureDirectories(context.getAnnotationProcessorPaths());
                 this.fileManager.setLocation(StandardLocation.ANNOTATION_PROCESSOR_PATH, context.getAnnotationProcessorPaths());
             }
         } catch (IOException e) {
             throw new RuntimeException("Cannot reset file manager", e);
         }
     }
+    
+    private void ensureDirectories(Iterable<File> directories){
+        for (File processorPath : directories) {
+            if (!processorPath.exists()) {
+                processorPath.mkdirs();
```

Comment:
Can you add a check for the return of `mkdirs`? It returns `false` when it fails, in which case you can throw a `RuntimeException` explaning the problem.